lib/forspell/cli.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+require 'optimist'
+require 'backports/2.5.0/hash/slice'
+require_relative 'runner'
+require_relative 'speller'
+require_relative 'reporter'
+require_relative 'loaders/file_loader'
+
+module Forspell
+  class CLI
+    ERROR_CODE = 2
+    CONFIG_PATH = File.join(Dir.pwd.to_s, '.forspell')
+    DEFAULT_CUSTOM_DICT = File.join(Dir.pwd.to_s, '.forspell.dict')
+
+    FORMATS = %w[readable yaml json].freeze
+    FORMAT_ERR = "must be one of the following: #{FORMATS.join(', ')}"
+
+    OPTION_KEYS = %i[
+      dictionary_name
+      custom_dictionaries
+      format
+      logfile
+      verbose
+      group
+    ].freeze
+
+    def self.call
+      init_options
+      create_files_list
+      init_speller
+      init_reporter
+      run
+    end
+
+    def self.create_files_list
+      @files = ARGV.flat_map do |path|
+        if File.directory?(path)
+          Loaders::FileLoader.new(path: path,
+                                  exclude_paths: @opts[:exclude_paths] || [])
+                             .process.result
+        else
+          [path]
+        end
+      end
```

Comment:
BTW, I feel like processing `exclude_paths` in FileLoader for each path separately is kinda wrong. Think this:
```
$ forspell *.md -e Changelog.md
```
Bash will expand this to `forspell README.md Contributing.md TODO.md Changelog.md -e Changelog.md`
...and what I've meant was "all `*.md` files in this folder, except for changelog (it has a lot of usernames and slang). 
You'll code will not process it correctly :)
Probably, `exclude_paths` should be handled ONCE, processing the entire list of files loaded from paths.